### PR TITLE
Remove WAL on safekeepers without s3 offloading.

### PR DESCRIPTION
**Pull request by arssher**
Until it is merged.

### CODE REVIEW

**Review by petuhovskiy — COMMENTED** [review]
It seems that we wanted to do that only on staging. Should we add some temp configuration to distinguish staging from production?

**Comment by arssher** [comment]
Hm. I hope s3 offloading lands before prod release, but let's do it optional anyway.

**Review by petuhovskiy — APPROVED** [review]
Ok, the next prod deployment is likely to be done by you, so you know what to fix before release :)

But option in config to trim without s3 still can be useful for capacity testing.

UPD: I missed the new changes before approval somehow, but I took a look and it's ok.

File: .circleci/ansible/systemd/safekeeper.service [hunk 1]
```diff
@@ -6,7 +6,7 @@ After=network.target auditd.service
 Type=simple
 User=safekeeper
 Environment=RUST_BACKTRACE=1 ZENITH_REPO_DIR=/storage/safekeeper/data LD_LIBRARY_PATH=/usr/local/lib
-ExecStart=/usr/local/bin/safekeeper -l {{ inventory_hostname }}.local:6500 --listen-http {{ inventory_hostname }}.local:7676 -p {{ first_pageserver }}:6400 -D /storage/safekeeper/data --broker-endpoints={{ etcd_endpoints }}
+ExecStart=/usr/local/bin/safekeeper -l {{ inventory_hostname }}.local:6500 --listen-http {{ inventory_hostname }}.local:7676 -p {{ first_pageserver }}:6400 -D /storage/safekeeper/data --broker-endpoints={{ etcd_endpoints }} --enable-s3-offload={{ safekeeper_enable_s3_offload }}
```
Comment (petuhovskiy) [inline .circleci/ansible/systemd/safekeeper.service:9]: What will happen to the old safekeeper when it'll get unknown option in the arguments? Will it ignore it silently?

Comment (arssher) [reply .circleci/ansible/systemd/safekeeper.service:9]: No, it will crash and burn :) Uh, I wanted to avoid programming in ansible.

Comment (arssher) [reply .circleci/ansible/systemd/safekeeper.service:9]: Hang on, how is it possible to get fresh service file without updating the binary?

**Comment by arssher** [comment]
Got on more test_pageserver_catchup failure, created issue on that
https://github.com/neondatabase/neon/issues/1666

**Pull request merged**